### ISSUE TITLE
Project needs maintainer

### ISSUE THREAD

**Issue by dead10ck** [spicavigo/kafka-rust#185]
Unfortunately, I have not had time the last few years to continue maintenance of this project, nor has @xitep or @spicavigo. Please post here if you are interested in becoming a maintainer.

**Comment by synlestidae** [comment]
@dead10ck Have you posted this to reddit or to the [rust users forum](https://users.rust-lang.org/)? Sad to see a big project like this go without a maintainer. I sent this issue to Confluent. Perhaps they could be interested.

**Comment by dead10ck** [comment]
I've had it posted in the Call for Participation section of the This Week in Rust blog in the past, and it didn't get any response. That was a while ago, though, so it may be worth posting again.

**Comment by gklijs** [comment]
Confluent most certainly isn't mostly because it's not using librdkafka, like all the other Confluent supported non-jvm clients do. I have to think about it very hard myself. It's an awesome project, but it's hard to keep up with all the features added. I know it will soak up a lot of time.

**Comment by johnward** [comment]
Are you still looking for a maintainer for this project I would be interested in taking it on, thanks John

**Comment by dead10ck** [comment]
Hi @johnward! Thank you for volunteering! It looks like I do not have admin access on the repo, so paging @xitep or @spicavigo.

**Comment by johnward** [comment]
OK great - what is the best way forward? thanks

**Comment by spicavigo** [comment]
I have sent an invitation to @johnward

**Comment by johnward** [comment]
Thank you, I will start tolook at some issues over the next few weeks.

**Comment by johnward** [comment]
I've started to tidy up the library. I am going through the examples over the next week or so. Then I will tackle the remaining issues. I use Kafka in my day job and see this as an important library for Rust.

**Comment by lerouxrgd** [comment]
Hello, this might be a bit late but I'd be interested in contributing.

Last year I worked a bit on a full Rust implementation of the Kafka protocol, code [here](https://github.com/lerouxrgd/kafka-rs)
The idea was to:
- Parse the [official spec](https://kafka.apache.org/protocol.html) with the [spec-parser](https://github.com/lerouxrgd/kafka-rs/tree/master/spec-parser) module (using a small [pest](https://pest.rs/) grammar)
- Then generate automatically all modules/structs/enum for the protocol in a [model.rs](https://github.com/lerouxrgd/kafka-rs/blob/master/kafka-protocol/src/model.rs) file
- Finally there are [serde](https://serde.rs/)-based [codecs](https://github.com/lerouxrgd/kafka-rs/tree/master/kafka-protocol/src/codec) that work with the auto generated model. The only drawback here is that I had to use [specialization feature](https://github.com/rust-lang/rust/issues/31844) which means use Rust nightly, but maybe it's doable to find a workaround.

I haven't re-generated the `models.rs` file for some time, so maybe some small tweaks are needed for the `spec-parser`, but it's simple to maintain.

I originally wanted to also develop an async client that would leverage this protocol implementation but I didn't have the time to get into it seriously.

Let me know whether you would be interested by this approach.

**Comment by johnward** [comment]
I would be interested in reviewing this approach. thanks

**Comment by Marwes** [comment]
I suppose I should point out a similar specification which I have worked on to and from https://github.com/Marwes/kafka-protocol .

Some notable distinctions

* It strives to be zero-copy as much as possible (definitely possible to take this further however).
* No `std::io::Read/Write`, just parsing from `&[u8]` which tends to be much faster
* The above, in conjunction with care to not use non-Send/Sync types means it is created with async in mind
* The generated parsers use https://github.com/Marwes/combine (could change, mostly just for simplicity of use atm)
* There is no multi version support atm, you just get one version of the API. I started on an approach where the parser and genereted structs support multiple versions of an API, rather than generating an enum variant for each version  as I believe that would be more useful, albeit much more complicated to implement.

All of these make for an API with less overhead at the expense of complexity
